### PR TITLE
Operation from lower scale

### PR DESCRIPTION
**Pull request by j3r3m1**
Code for basic statistical operations (sum and average) from a scale to another (building to block, building to rsu or block to rsu).

Useful for indicators such as: block_building_volume, block_building_area, block_floor_area, etc.

Another indicator will be used for weighted statistical operations.

### CODE REVIEW

File: geoindicators/src/main/groovy/org/orbisgis/BlockIndicators.groovy [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+@BaseScript Geoclimate geoclimate
+
+/**
+ * This process is used to compute basic statistical operations on a specific variable from a lower scale (for
+ * example the sum of each building volume constituting a block to calculate the block volume)
+ * @return A database table name.
+ * @author Erwan Bocher
+ */
+static IProcess unweightedOperationFromLowerScale() {
```
Comment (ebocher) [inline geoindicators/src/main/groovy/org/orbisgis/BlockIndicators.groovy]: aggregatedStatistics ?

File: geoindicators/src/main/groovy/org/orbisgis/BlockIndicators.groovy [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+return processFactory.create(
+        "Unweighted statistical operations from lower scale",
+        [inputLowerScaleTableName: String,inputCorrelationTableName: String,inputIdLow: String, inputIdUp: String,
+         inputToTransfo: String,operations: String[], outputTableName: String, datasource: JdbcDataSource],
+        [outputTableName : String],
+        { inputLowerScaleTableName, inputCorrelationTableName, inputIdLow, inputIdUp,
+          inputToTransfo, operations, outputTableName, datasource ->
+            def ops = ["SUM","AVG"]
```
Comment (ebocher) [inline geoindicators/src/main/groovy/org/orbisgis/BlockIndicators.groovy]: count, min, max ???

File: geoindicators/src/main/groovy/org/orbisgis/BlockIndicators.groovy (hunk 1 above)
Comment (j3r3m1) [reply geoindicators/src/main/groovy/org/orbisgis/BlockIndicators.groovy]: unweightedAggregatedStatistics in order to differentiate from the weighted ones ?

File: geoindicators/src/main/groovy/org/orbisgis/BlockIndicators.groovy (hunk 2 above)
Comment (j3r3m1) [reply geoindicators/src/main/groovy/org/orbisgis/BlockIndicators.groovy]: So far (for the indicators we have defined) there is no need for such functions. However, we could add them if you think it is necessary for the future.

File: geoindicators/src/main/groovy/org/orbisgis/BuildingIndicators.groovy [hunk 3]
```diff
@@ -222,6 +222,116 @@ static IProcess buildingFormProperties() {
...
+ * This process extract the building closest distance to an other building. A buffer of defined size (bufferDist argument)
+ * is used to get the buildings within the building of interest and then the minimum distance is calculated.
+ *
+ * @return A database table name.
+ * @author Jérémy Bernard
+ */
+static IProcess buildingMinimumBuildingSpacing() {
+    return processFactory.create(
+            "Building minimum building spacing",
+            [inputBuildingTableName: String,inputFields:String[],bufferDist: Integer
```
Comment (ebocher) [inline geoindicators/src/main/groovy/org/orbisgis/BuildingIndicators.groovy]: double ?

**Pull request merged**